apps/api_web/lib/api_web/event_stream/supervisor.ex

```diff
@@ -2,52 +2,74 @@ defmodule ApiWeb.EventStream.Supervisor do
   @moduledoc """
   Supervisor for the infrastructure needed for event streaming.
 
-  - ServerRegistry - mapping controller/params pairs to an EventStream.Server
+  - ServerRegistry - mapping controller/params pairs to a DiffServer
   - ServerSupervisor - DynamicSupervisor for managing the children
   """
+
+  alias ApiWeb.EventStream.{DiffServer, ServerRegistry, ServerSupervisor}
+
   def start_link do
     Supervisor.start_link(
       [
-        {Registry, name: ApiWeb.EventStream.ServerRegistry, keys: :unique},
-        {DynamicSupervisor, name: ApiWeb.EventStream.ServerSupervisor, strategy: :one_for_one}
+        {Registry, name: ServerRegistry, keys: :unique},
+        {DynamicSupervisor, name: ServerSupervisor, strategy: :one_for_one}
       ],
       strategy: :one_for_all
     )
   end
 
   @doc """
-  Returns a {:ok, pid} tuple for an EventStream.Server for the given connection/module.
-
-  If one already exists, the same one will be returned.
+  Subscribes the current process to the appropriate DiffServer for the given connection/module,
+  starting it if it doesn't exist.
   """
-  def server_child(conn, module) do
+  @spec server_subscribe(Plug.Conn.t(), module) :: DynamicSupervisor.on_start_child()
+  def server_subscribe(conn, module) do
     case start_child(conn, module) do
       {:ok, pid} ->
-        :ok = ApiWeb.EventStream.DiffServer.subscribe(pid)
+        :ok = DiffServer.subscribe(pid)
         {:ok, pid}
 
       {:error, {:already_started, pid}} ->
-        :ok = ApiWeb.EventStream.DiffServer.subscribe(pid)
+        :ok = DiffServer.subscribe(pid)
         {:ok, pid}
 
       {:error, _} = error ->
         error
     end
   end
 
-  @doc "Unsubscribes the current pid from the given Server pid."
-  defdelegate server_unsubscribe(pid), to: ApiWeb.EventStream.DiffServer, as: :unsubscribe
+  @doc "Unsubscribes the current process from the given DiffServer."
+  @spec server_unsubscribe(pid) :: :ok
+  def server_unsubscribe(pid) do
+    :ok = DiffServer.unsubscribe(pid)
+  end
```

Comment:
Is there a reason to re-write this, rather than delegating?

Reply:
Ah, no, there isn't a reason to rewrite it anymore. This is a remnant of the previous iteration where this function started doing more things (with the connection registry).